name.abuchen.portfolio/src/name/abuchen/portfolio/math/Risk.java

```diff
@@ -0,0 +1,73 @@
+package name.abuchen.portfolio.math;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.DoubleStream;
+
+public final class Risk
+{
+
+    public static double calculateMaxDrawdown(long[] values)
+    {
+        Double peak = Double.MIN_VALUE;
+        List<Double> dd = new ArrayList<Double>();
+        Double maxDD = 0d;
+        int i = 0;
+        for (double value : values)
+        {
+            peak = Math.max(peak, value);
+            dd.add(i, (peak - values[i]) / peak);
+            maxDD = Math.max(maxDD, dd.get(i));
+            i++;
+        }
+        return maxDD;
+    }
+
+    private static double[] getReturns(long[] values)
+    {
+        double[] returns = new double[values.length - 1];
+        for (int i = 0; i < returns.length; i++)
+        {
+            returns[i] = (double) values[i + 1] / values[i];
+        }
+        return returns;
+    }
+
+    public static double calculateAverageVolatility(long[] values)
+    {
+        double[] vola = Risk.getReturns(values);
```

Comment:
Bei der "average volatility" sehe ich zwei Probleme:

**Erstens** sollte sollte die Returns nicht auf Grund der "totals" berechnet werden. Neben wir an, man zahlt 10K auf das Konto ein. Dann ändert sich der Gesamtwert. Das ist aber keine Volatilität des Portfolios, das ist performanceneutral. Meinem Verständnis nach sollte man hier "accumulated" nehmen. Das sollte doch die "returns" des Portfolios sein. Was meinst Du?

**Zweitens** berechne ich die Wert täglich - also auch für den Samstag und Sonntag. Das sind aber keine Handelstage, d.h. normalerweise ändert sich die Performance nicht (es sei denn, man erfasst eine Einzahlung für einen Samstag). Wenn man jetzt die Volatilität auf Basis aller Tage berechnet, dann unterschätzt man die tatsächliche Volatiltität weil sich Samstag und Sonntag nix ändert. Das müsste man eventuell ausrechnen.

Das gilt im Übrigen auch für Handelstage. Und die können ja auch für die Börsen in Deutschland/Europa/USA unterschiedlich sein. Wie würde man die Volatilität für ein Portfolio berechnen, dass Papiere an verschiedenen Börsen gehandelt wird.